src/sentry/api/endpoints/auth_index.py

```diff
@@ -257,6 +259,14 @@ def delete(self, request: Request, *args, **kwargs) -> Response:
 
         Deauthenticate all active sessions for this user.
         """
+        # For signals to work here, we must promote the request.user to a full user object
+        request.user = _promote_request_api_user(request)
```

Comment:
This, however, concerns me.  Ideally, we do not have wildly semantically different types on the same attribute of the request.  Is it possible for us to change the way those signals work to _not_ assume `request.user` is an ORM object?

Think about it--- if those signals really do need an ORM object, we're admitting that there's tons of code that must run in the control silo, since obv they won't work in the region silo.

Reply:
Even though it's onerous, I think it is important to not do this wrong -- let's make a deeper refactor and make sure `request.user` does not need to be an ORM object.

Reply:
also, I'm willing to grant time to pair and take a closer look at this.  I think I could provide a more practical solution without huge work if I can see all the related code and concerns which aren't in the PR itself.

Reply:
We should do that then. It _might_ work to just update or add another receiver here: https://github.com/getsentry/sentry/blob/master/src/sentry/models/user.py#L418. Have to dig through some django code to make sure.